File: src/main/java/br/com/g1bet/model/Time.java
package br.com.g1bet.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.List;

@Entity
public class Time {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Size(min = 5, max = 100)
    private String nome;

    @OneToMany(mappedBy = "timeCasa", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Partida> partidasCasa;

    @OneToMany(mappedBy = "timeVisitante", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Partida> partidasVisitante;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }
}

File: src/main/java/br/com/g1bet/model/Time.kt
package br.com.g1bet.model

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
class Time {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @NotNull
    @Size(min = 5, max = 100)
    var nome: String? = null

    @OneToMany(mappedBy = "timeCasa", cascade = arrayOf(CascadeType.ALL))
    @JsonIgnore
    private val partidasCasa: List<Partida>? = null

    @OneToMany(mappedBy = "timeVisitante", cascade = arrayOf(CascadeType.ALL))
    @JsonIgnore
    private val partidasVisitante: List<Partida>? = null
}